File: src/main/java/net/benwoodworth/fastcraft/core/api/dependencies/recipes/RecipeSource.java
package net.benwoodworth.fastcraft.core.api.dependencies.recipes;

import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * A source of recipes. (e.g. from plugins that register custom recipes)
 *
 * @param <TItem> the native item type
 */
public interface RecipeSource<TItem> {

    /**
     * Get this recipe source's crafting recipes.
     *
     * @return the crafting recipes
     */
    @NotNull
    List<FcCustomRecipe<TItem>> getCraftingRecipes();
}

File: src/main/kotlin/net/benwoodworth/fastcraftplus/core/recipe/RecipeSource.kt
package net.benwoodworth.fastcraftplus.core.recipe

/**
 * A source of recipes. (Bukkit, Sponge, plugins, etc.)
 */
interface RecipeSource<TRecipe : FcRecipe<*>> {

    /**
     * Get this recipe source's crafting recipe.
     */
    fun getCraftingRecipes(): Array<TRecipe>
}